assign interface object

The following program `sample.go` triggers a panic:

```go
package main

type I1 interface {
    Truc()
}   

type T1 struct{}

func (T1) Truc() { println("in T1 truc") }

var x I1 = T1{}

func main() {
    println("ok")
}   
```

Expected result:
```console
$ go run ./sample.go
ok
```

Got:
```console
$ yaegi ./sample.go
./sample.go:11:12: panic
panic: reflect.Set: value of type struct {} is not assignable to type interp.valueInterface [recovered]
        panic: reflect.Set: value of type struct {} is not assignable to type interp.valueInterface

goroutine 1 [running]:
github.com/containous/yaegi/interp.runCfg.func1(0xc000272000, 0xc000288000)
        /home/marc/go/src/github.com/containous/yaegi/interp/run.go:112 +0x218
panic(0xc99a00, 0xc000172700)
        /usr/lib/go/src/runtime/panic.go:679 +0x1b2
reflect.Value.assignTo(0xccfde0, 0x155be40, 0x199, 0xdd7d72, 0xb, 0xd200c0, 0x0, 0xc865e0, 0x0, 0x155be40)
        /usr/lib/go/src/reflect/value.go:2403 +0x432
reflect.Value.Set(0xd200c0, 0xc0001767c0, 0x199, 0xccfde0, 0x155be40, 0x199)
        /usr/lib/go/src/reflect/value.go:1537 +0xbd
github.com/containous/yaegi/interp.compositeLit.func1(0xc000272000, 0x153e020)
        /home/marc/go/src/github.com/containous/yaegi/interp/run.go:1525 +0x2c3
github.com/containous/yaegi/interp.runCfg(0xc000288000, 0xc000272000)
        /home/marc/go/src/github.com/containous/yaegi/interp/run.go:118 +0x7a
github.com/containous/yaegi/interp.(*Interpreter).run(0xc0002686e0, 0xc000284000, 0x0)
        /home/marc/go/src/github.com/containous/yaegi/interp/run.go:96 +0x239
github.com/containous/yaegi/interp.(*Interpreter).Eval(0xc0002686e0, 0xc00027a000, 0x9b, 0x29b, 0xc00027a000, 0x9b, 0x0, 0x1)
        /home/marc/go/src/github.com/containous/yaegi/interp/interp.go:357 +0x3f3
main.main()
        /home/marc/go/src/github.com/containous/yaegi/cmd/yaegi/yaegi.go:82 +0x42d
```

--- a/_test/interface12.go
+++ b/_test/interface12.go
@@ -1,0 +1,18 @@
+package main
+
+type I1 interface {
+	Truc()
+}
+
+type T1 struct{}
+
+func (T1) Truc() { println("in T1 truc") }
+
+var x I1 = T1{}
+
+func main() {
+	x.Truc()
+}
+
+// Output:
+// in T1 truc

--- a/interp/run.go
+++ b/interp/run.go
@@ -1495,6 +1495,15 @@ func compositeBinStruct(n *node) {
 	}
 }
 
+func destType(n *node) *itype {
+	switch n.anc.kind {
+	case assignStmt, defineStmt:
+		return n.anc.child[0].typ
+	default:
+		return n.typ
+	}
+}
+
 // compositeLit creates and populates a struct object
 func compositeLit(n *node) {
 	value := valueGenerator(n, n.findex)
@@ -1503,6 +1512,7 @@ func compositeLit(n *node) {
 	if !n.typ.untyped {
 		child = n.child[1:]
 	}
+	destInterface := destType(n).cat == interfaceT
 
 	values := make([]func(*frame) reflect.Value, len(child))
 	for i, c := range child {
@@ -1519,9 +1529,12 @@ func compositeLit(n *node) {
 		for i, v := range values {
 			a.Field(i).Set(v(f))
 		}
-		if d := value(f); d.Type().Kind() == reflect.Ptr {
+		switch d := value(f); {
+		case d.Type().Kind() == reflect.Ptr:
 			d.Set(a.Addr())
-		} else {
+		case destInterface:
+			d.Set(reflect.ValueOf(valueInterface{n, a}))
+		default:
 			d.Set(a)
 		}
 		return next

